crates/mock/src/async_client.rs

```diff
@@ -0,0 +1,60 @@
+use crate::utils::AccountInfo;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use solana_sdk::genesis_config::ClusterType;
+use surf::Client;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RpcResponse<T> {
+    pub jsonrpc: String,
+    pub result: T,
+    pub id: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ResultData {
+    pub context: Context,
+    pub value: AccountInfo,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Context {
+    #[serde(rename = "apiVersion")]
+    pub api_version: String,
+    pub slot: u64,
+}
+
+pub async fn fetch_account_async(
+    cluster: ClusterType,
+    account_pubkey: &str,
+) -> Result<AccountInfo, Box<dyn std::error::Error>> {
+    let rpc_url = match cluster {
+        ClusterType::MainnetBeta => "https://api.mainnet-beta.solana.com".to_string(),
+        ClusterType::Devnet => "https://api.devnet.solana.com".to_string().to_string(),
+        ClusterType::Testnet => "https://api.testnet.solana.com".to_string(),
+        _ => "https://api.mainnet-beta.solana.com".to_string(),
+    };
```

Comment:
Thinking on this I'd like the this to be read off config since larger teams will have private RPC. It could be a follow up and something for you to think about @ray-kast related to Vixen config.

Reply:
Make sense we can ask users to create a .env or something or else toml also works